proposed/2021/NuGet.orgTFMs.md

```diff
@@ -0,0 +1,96 @@
+
+### Success: How do we know if we’ve solved this problem? 
+
+- By measuring the change in the percentage of successful installs and lowering the amount of [NU1202](https://docs.microsoft.com/en-us/nuget/reference/errors-and-warnings/nu1202) errors in telemetry. 
+- By measuring the usage of target framework & supported platform filters on NuGet.org & Visual Studio. 
+- By gauging excitement/disappointment after blogging about this feature & analyzing sentiment on Twitter/GitHub/DevCom/etc. 
+
+### Audience: Who are we building for? 
+
+We are building this for the .NET developer who is browsing & making decisions about including a NuGet package into their project based on target framework & supported platform criteria. 
+
+## Explanation
+
+### Functional Explanation
+
+#### Context and Scope 
+When a developer accesses the details page for a NuGet package, they will see information on target frameworks in a dropdown menu. The framework information will be surfaced from the package. In cases where versions are not specified (such as .net) or other unexpected scenarios, ‘Version Not Specified’ or similar messages will be surfaced. 
+
+##### Minimal Requirements 
+- List of Frameworks on NuGet.org
+
+##### Non-Requirements 
+- List of Platforms on NuGet.org
+  - Users interviewed did not express need for supported platform information 
+  - Cannot be surfaced directly from package data and would require additional layer of soft calculation from engineering (no clear mapping scheme currently exists between target frameworks and supported platforms) 
+  - Can be revisited in future 
+- Filters for Frameworks on NuGet.org
+    - Users interviewed were ambivalent on filters, requires further investigation. Can be revisited in future 
+- Filters for Platforms on NuGet.org
+  - Users interviewed were ambivalent on filters, requires further investigation. Can be revisited in future 
+
+Although Target Frameworks & Supported Platforms are relatively similar in general contexts and often have overlap, we must clearly define each of these items based on their differences. Today we live in an ecosystem where Target Frameworks can represent both a supported platform, or a subset of supported platforms. 
+
+**Target Framework** – A target framework is a specification of the APIs available to an application or library. 
+
+Example: .NET Framework, .NET Core, .NET Standard, .NET 5 are target frameworks. 
+
+**Supported Platform** – A supported platform is a target that the APIs can be ran on. 
+
+Example: iOS, Android, Windows are platforms. 
+
+### Solution 
+
+- A .NET developer can view Frameworks of a package on NuGet.org: 
+![](../../meta/resources/NuGet.orgTFMs/PackageDetailsWithTFMs.png)
```

Comment:
Consider a display mimicking the API docus:
https://docs.microsoft.com/en-us/dotnet/api/system.text.stringbuilder?view=net-5.0#applies-to
There's also Immo's https://apisof.net/catalog/e40ef6ad-bd6e-9f29-7864-b942e37be11f

Reply:
Looks great. Going back to framework precedence order it by "Latest TFM -> Oldest least compatible TFM"

Reply:
https://docs.microsoft.com/en-us/dotnet/standard/frameworks#latest-versions in other words

Reply:
Are we leaving our intended MVP scope behind a little? This would be impressive and might be a good v2 feature, but I'm concerned that we're throwing some cost in here where we don't need to. Can we backlog this one?

Reply:
I don't think there's extra cost here. Just deciding on whether a format like the existing `Dependencies` section or a table like `Version History` would represent this information better given NuGet.org is already quite heavy on vertical real estate.

Tables have pros in terms of saving vertical space at the cost of horizontal space.

Headers have pros in terms of saving horizontal space at the cost of vertical space.

/cc @jcjiang

Reply:
Let's step back a moment. This seems predicated on us not just listing actual TFMs present in the archive (as originally floated), but on exploding this to every possible supported TFM. I want to be sure that we're intentionally adding this to the spec, because there's cost there. That's my cost concern--not <br /> vs. <tr /> tags.

Reply:
If this is to be added to the spec, we'd need to call it out explicitly, and I'd suggest either:
- put it in as a nice-to-have, not an MVP item, or
- backlog it for the future

Reply:
If we put the technical cost aside since a user could multitarget the world & have every single TFM in their archive

i.e. Microsoft.CSharp

![image](https://user-images.githubusercontent.com/1288848/108278866-4dfa6300-7141-11eb-9f10-b0dbbe1fb3ad.png)

This gets rather unwieldy quickly in terms of vertical space.

Reply:
This example wouldn't help much with a table as only the `netstandard` items would likely save space. This one is just unique 😂

Reply:
Should just resolve this feedback & move on with the original mockup if it's easiest! Maybe put under a `Unresolved Questions` if @jcjiang wanted to dive more in the UX w/ the UX team here.

Reply:
Yep, that works for me if @jcjiang is good with that approach. Good example! It's always good to look at nth degree cases like this.

Reply:
I've added this as an unresolved question.